### PR TITLE
fix(launch): support ssh git urls and submodules in agent

### PR DESCRIPTION
**Pull request by KyleGoyette**
Fixes
-----

[Fixes WB-9993](https://wandb.atlassian.net/browse/WB-9993)

Fixes #NNNN

Description
-----------

Fixes our git fetch system to use https even if the job uses an SSH repo by using Jamie's github_reference

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at af57a50</samp>

> _Sing, O Muse, of the skillful coder who refined the codebase_
> _With clever changes to the `fetch` method and the `_fetch_git_repo` function_
> _He made the `GitHubReference` class more powerful and versatile_
> _Like Hephaestus, who forged the weapons of the gods and heroes_

### CODE REVIEW

File: wandb/apis/public.py [hunk 1]
```diff
@@ -4356,7 +4356,7 @@ def _configure_launch_project_repo(self, launch_project):
         git_info = self._job_info.get("source", {}).get("git", {})
         _fetch_git_repo(
             launch_project.project_dir,
-            git_info["remote"],
+            "git@github.com:wandb/launch-jobs.git",  # git_info["remote"],
```
Comment (jamie-rasmussen) [inline wandb/apis/public.py:4359]: I think you might not have meant to leave this in?

File: wandb/sdk/launch/errors.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    pass
+
+
+class ExecutionError(Error):
+    """Generic execution exception."""
+
+    pass
+
+
+class SweepError(Error):
```
Comment (jamie-rasmussen) [inline wandb/sdk/launch/errors.py:22]: Is this unused?

File: wandb/sdk/launch/github_reference.py [hunk 3]
```diff
@@ -194,6 +194,7 @@ def fetch(self, dst_dir: str) -> None:
         # TODO: Is there a better way to do this?
         default_branch = None
         if not commit and not branch:
+            print(repo, repo.references)
```
Comment (jamie-rasmussen) [inline wandb/sdk/launch/github_reference.py:197]: Remove

File: wandb/sdk/launch/errors.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/errors.py:22]: This does appear unused but I haven't added it, I just moved it

File: wandb/apis/public.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/apis/public.py:4359]: Yep, that was for testing, didn't mean to commit it

**Review by jamie-rasmussen — APPROVED** [review]

**Pull request merged**